elks/arch/i86/kernel/process.c

```diff
@@ -92,11 +92,10 @@ unsigned get_ustack(register struct task_struct *t,int off)
  */
 void arch_setup_kernel_stack(register struct task_struct *t)
 {
-    put_ustack(t, -2, USER_FLAGS);			/* Flags */
-    put_ustack(t, -4, (int) current->t_regs.cs);	/* user CS */
-    put_ustack(t, -6, 0);				/* addr 0 */
+    put_ustack(t, -2, USER_FLAGS);		/* Flags */
+    put_ustack(t, -4, (int) t->t_regs.cs);	/* user CS */
+    put_ustack(t, -6, 0);			/* addr 0 */
     t->t_regs.sp -= 6;
-    t->t_kstackm = KSTACK_MAGIC;
```

Comment:
OK, got it, thanks Juan. The marker is already set in `find_empty_process()`. Maybe for the point 3. that task structure initialization should be moved out of this function ?